e2e/nomostest/wait_for_sync.go

```diff
@@ -233,86 +233,47 @@ func (nt *NT) WatchForSync(
 }
 
 // WaitForRootSyncSourceError waits until the given error (code and message) is present on the RootSync resource
-func (nt *NT) WaitForRootSyncSourceError(rsName, code string, message string, opts ...WaitOption) {
+func (nt *NT) WaitForRootSyncSourceError(rsName, code string, message string) error {
 	nt.T.Helper()
-	Wait(nt.T, fmt.Sprintf("RootSync %s source error code %s", rsName, code), nt.DefaultWaitTimeout,
-		func() error {
-			nt.T.Helper()
-			rs := k8sobjects.RootSyncObjectV1Beta1(rsName)
-			if err := nt.KubeClient.Get(rs.GetName(), rs.GetNamespace(), rs); err != nil {
-				return err
-			}
-			// Only validate the rendering status, not the Syncing condition
-			// TODO: Remove this hack once async sync status updates are fixed to reflect only the latest commit.
-			return testpredicates.ValidateError(rs.Status.Source.Errors, code, message, nil)
-			// syncingCondition := rootsync.GetCondition(rs.Status.Conditions, v1beta1.RootSyncSyncing)
-			// return validateRootSyncError(rs.Status.Source.Errors, syncingCondition, code, message, []v1beta1.ErrorSource{v1beta1.SourceError})
-		},
-		opts...,
-	)
+	predicates := []testpredicates.Predicate{
+		testpredicates.RootSyncHasSourceError(code, message),
+		testpredicates.RootSyncHasErrorSourceRefs([]v1beta1.ErrorSource{v1beta1.SourceError}),
+	}
+	return nt.Watcher.WatchObject(kinds.RootSyncV1Beta1(), rsName, configsync.ControllerNamespace,
+		testwatcher.WatchPredicates(predicates...))
 }
 
 // WaitForRootSyncSyncError waits until the given error (code and message) is present on the RootSync resource
-func (nt *NT) WaitForRootSyncSyncError(rsName, code string, message string, resources []v1beta1.ResourceRef, opts ...WaitOption) {
+func (nt *NT) WaitForRootSyncSyncError(rsName, code string, message string, resources []v1beta1.ResourceRef) error {
```

Comment:
Since it's using a watch now, can you rename it WatchFor- and move it into the testwatcher package? Trying to reduce NT methods.

Reply:
done